core/src/main/java/org/openstack4j/openstack/artifact/internal/BaseArtifactServiceImpl.java

```diff
@@ -36,27 +37,27 @@ protected BaseArtifactServiceImpl(ArtifactType artifactType) {
 
     protected <T> T list(Class<T> clazz) {
 
-        return get(clazz,uri("/artifacts/"+ artifactType.value())).execute();
+        return get(clazz,uri(ClientConstants.PATH_ARTIFACTS+ClientConstants.URI_SEP+ artifactType.value())).execute();
```

Comment:
BaseArtifactServiceImpl.java is having generic implementation. ToscaTemplatesArtifactServiceImpl.java is extending BaseArtifactServiceImpl. Similary there will be different impl classes extending BaseArtifactServiceImpl. Hence generics is used.

Other comments i will take it.